libpromises/attributes.c

```diff
@@ -364,6 +364,7 @@ ExecContain GetExecContainConstraints(const EvalContext *ctx, const Promise *pp)
     e.owner = PromiseGetConstraintAsUid(ctx, "exec_owner", pp);
     e.group = PromiseGetConstraintAsGid(ctx, "exec_group", pp);
     e.preview = PromiseGetConstraintAsBoolean(ctx, "preview", pp);
+    e.have_nooutput = PromiseBundleOrBodyConstraintExists(ctx, "no_output", pp);
```

Comment:
Why do we need `have_nooutput`? Can't we just set the default for `no_output` here no matter if the attribute exists, and overwrite it with the given value if is indeed given?